src/inputs/Autocomplete.tsx

```diff
@@ -0,0 +1,136 @@
+import { Key, ReactNode, useRef } from "react";
+import { useComboBox, useOverlayPosition } from "react-aria";
+import { Item, useComboBoxState } from "react-stately";
+import { Icon } from "src/components";
+import { Popover } from "src/components/internal";
+import { PresentationFieldProps } from "src/components/PresentationContext";
+import { ListBox } from "src/inputs/internal/ListBox";
+import { TextFieldBase, TextFieldBaseProps } from "src/inputs/TextFieldBase";
+import { valueToKey } from "src/inputs/Value";
+
+interface AutocompleteProps<T>
+  extends Pick<PresentationFieldProps, "labelStyle">,
+    Pick<TextFieldBaseProps<any>, "label" | "clearable" | "startAdornment"> {
+  onSelect: (item: T) => void;
+  /** A function that returns how to render the an option in the menu. If not set, `getOptionLabel` will be used */
+  getOptionMenuLabel?: (o: T) => ReactNode;
+  /** A function that returns the string value of the option. Used for accessibility purposes */
+  getOptionLabel: (o: T) => string;
+  /** A function that returns a unique key for an option */
+  getOptionValue: (o: T) => string;
+  /** Called when the input value changes */
+  onChange: (value: string) => void;
```

Comment:
Suggestion: Maybe `onInputChange` or `onTypeAheadChange`? Just thinking to clarify it's the "hey they typed something different, redo your GQL call" callback and not the "they actually picked option #4".

Reply:
Yup. Adding that specificity into the prop name would be helpful.